pkg/systemd/services/services.jsx

```diff
@@ -399,120 +433,71 @@
-                                        .finally(() => {
-                                            // Remove units from state that are not listed from the API in this iteration
-                                            const unit_by_path = Object.assign({}, this.state.unit_by_path);
-                                            let hasExtraEntries = false;
-                                            const newState = {};
-
-                                            for (const unitPath in this.state.unit_by_path) {
-                                                if (!this.seenPaths.has(unitPath)) {
-                                                    hasExtraEntries = true;
-                                                    delete unit_by_path[unitPath];
-                                                    Object.keys(this.path_by_id).forEach(id => {
-                                                        if (this.path_by_id[id] == unitPath)
-                                                            delete this.path_by_id[id];
-                                                    });
-                                                }
-                                            }
-                                            if (hasExtraEntries)
-                                                newState.unit_by_path = unit_by_path;
-
-                                            newState.isFullyLoaded = true;
-                                            this.props.setIsLoading(false);
-
-                                            this.setState(newState);
-                                            this.processFailedUnits();
-                                        });
-                            }, ex => {
-                                this.props.SetIsLoading(false);
-                                this.setState({ error: cockpit.format(_("Listing unit files failed: $0"), ex.toString()) });
-                            });
-                }, ex => {
-                    this.props.setIsLoading(false);
-                    this.setState({ error: cockpit.format(_("Listing units failed: $0"), ex.toString()) });
-                });
+                    this.units = units;
+                    this.unit_files = unit_files;
+                    this.processFailedUnits();
+
+                    return Promise.all(timerPromises);
+                })
+                .then((timerResults) => {
+                    for (let i = 0; i < timerResults.length; i++) {
+                        const [timer_props] = timerResults[i];
+                        const unit = {};
+                        this.addTimerProperties(timer_props, unit);
+                        this.timers[timerPaths[i]] = unit;
+                    }
+
+                    this.setState({ isFullyLoaded: true });
+                })
+                .catch(ex => this.setState({ error: cockpit.format(_("Listing units failed: $0"), ex.toString()) }))
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18422-20230307-120556-7e7cb6b6-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)